06-building-a-program-start-to-finish/writing-tests/output/spec/dplex-test-spec.js

```diff
@@ -0,0 +1,205 @@
+    });
+  });
+
+  describe('Tokenize words', () => {
+    it('should strip out special characters from excerpt in documents', () => {
+      let excerpt = 'Alice l##$oves her ima&&gination?';
+      const expectedTokens = ['alice', 'loves', 'her', 'imagination'];
+      excerpt = this.indexInstance.tokenize(excerpt);
+      expect(excerpt).toEqual(expectedTokens);
+    });
+  });
+
+  describe('Search index', () => {
+    it('should return false for an empty String', () => {
+      const term = '';
+      expect(this.indexInstance.searchIndex(term, 'books'))
+      .toBeFalsy();
+    });
+  });
+
+  describe('Get index', () => {
+    it('should return false for an empty filename', () => {
+      const filename = 'books';
+      const expectedOutput = { alice: [0],
+        falls: [0],
+        into: [0],
+        a: [0, 1, 2],
+        rabbit: [0],
+        hole: [0],
+        and: [0, 1, 2],
+        enters: [0],
+        world: [0],
+        full: [0],
+        of: [0, 1, 2],
+        imagination: [0],
+        in: [0],
+        wonderland: [0],
+        an: [1, 2],
+        unusual: [1, 2],
+        alliance: [1, 2],
+        man: [1, 2],
+        elf: [1, 2],
+        dwarf: [1, 2],
+        wizard: [1, 2],
+        hobbit: [1, 2],
+        seek: [1, 2],
+        to: [1, 2],
+        destroy: [1, 2],
+        powerful: [1, 2],
+        ring: [1, 2],
+        the: [1, 2],
+        lord: [1, 2],
+        rings: [1, 2],
+        fellowship: [1, 2] };
+      this.indexInstance.createIndex(this.books, 'books');
+      expect(this.indexInstance.getIndex(filename))
+        .toEqual(expectedOutput);
+    });
+  });
+
+  describe('Get index', () => {
+    it('should return the appropriate output for the given filename', () => {
+      const filename = '';
+      this.indexInstance.createIndex(this.books, 'books');
+      expect(this.indexInstance.getIndex(filename))
+        .toBeFalsy();
+    });
+  });
+
+  describe('Validate File', () => {
+    it('should return false for incorrect document structure', () => {
+      const term = { t1: 'Welcome home', text: 'This is really home' };
+      expect(this.indexInstance.validateFile(term)).toBeFalsy();
+    });
+  });
+
+  describe('Create Index Data', () => {
+    it('should return false for incorrect document structure', () => {
+      const term = { t1: 'Welcome home', text: 'This is really home' };
+      expect(this.indexInstance.createIndex(term, 'term')).toBeFalsy();
+    });
+  });
+
+  describe('Get Document Data', () => {
+    it('should return the approriate object for a given document',
+    () => {
+      const expectedOutput = { documentNum: 0,
+        textTokens: ['welcome', 'this', 'is', 'a', 'test', 'document'] };
+      const documentNum = 0;
+      const term = [{ text: 'Welcome', title: 'This is a test document' }];
+      expect(this.indexInstance
+        .getDocumentTokens(term, documentNum)).toEqual(expectedOutput);
+    });
+  });
+});
```

Comment:
Yo clearly have not implemented the feedback you had received. Please do that and then I will have another look .